Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Home.kt

```diff
@@ -67,107 +64,139 @@ import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.util.lerp
 import androidx.core.os.ConfigurationCompat
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavController
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.compose.KEY_ROUTE
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.getBackStackEntry
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.rememberNavController
 import com.example.jetsnack.R
-import com.example.jetsnack.ui.components.JetsnackScaffold
 import com.example.jetsnack.ui.components.JetsnackSurface
 import com.example.jetsnack.ui.home.cart.Cart
 import com.example.jetsnack.ui.home.search.Search
 import com.example.jetsnack.ui.theme.JetsnackTheme
-import com.google.accompanist.insets.ProvideWindowInsets
 import com.google.accompanist.insets.navigationBarsPadding
 
-@Composable
-fun Home(onSnackSelected: (Long) -> Unit) {
-    val (currentSection, setCurrentSection) = rememberSaveable {
-        mutableStateOf(HomeSections.Feed)
+fun NavGraphBuilder.home(
+    onSnackSelected: (Long, NavBackStackEntry) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    composable(HomeSections.FEED.route) { from ->
+        Feed(onSnackClick = { id -> onSnackSelected(id, from) }, modifier)
     }
-    val navItems = HomeSections.values().toList()
-    JetsnackScaffold(
-        bottomBar = {
-            JetsnackBottomNav(
-                currentSection = currentSection,
-                onSectionSelected = setCurrentSection,
-                items = navItems
-            )
-        }
-    ) { innerPadding ->
-        val modifier = Modifier.padding(innerPadding)
-        Crossfade(currentSection) { section ->
-            when (section) {
-                HomeSections.Feed -> Feed(
-                    onSnackClick = onSnackSelected,
-                    modifier = modifier
-                )
-                HomeSections.Search -> Search(onSnackSelected, modifier)
-                HomeSections.Cart -> Cart(onSnackSelected, modifier)
-                HomeSections.Profile -> Profile(modifier)
-            }
-        }
+    composable(HomeSections.SEARCH.route) { from ->
+        Search(onSnackClick = { id -> onSnackSelected(id, from) }, modifier)
+    }
+    composable(HomeSections.CART.route) { from ->
+        Cart(onSnackClick = { id -> onSnackSelected(id, from) }, modifier)
     }
+    composable(HomeSections.PROFILE.route) {
+        Profile(modifier)
+    }
+}
+
+private object HomeDestinations {
```

Comment:
Nit: Is this object necessary? I've been inlining the routes into my equivalent of `HomeSections` and it works well: https://github.com/chrisbanes/tivi/blob/main/app/src/main/java/app/tivi/AppNavigation.kt

Reply:
Fixed